src-built-in/components/appLauncher2/src/stores/StoreActions.js

```diff
@@ -191,11 +192,16 @@ function addApp(app = {}, cb) {
 	}
 	data.folders[MY_APPS].apps[appID] = data.apps[appID]
 	data.folders[folder].apps[appID] = data.apps[appID]
-	// Save appDefinitions and then folders
-	_setValue('appDefinitions', data.apps, () => {
-		_setFolders()
-		cb && cb()
-	})
+	FSBL.Clients.LauncherClient.addUserDefinedComponent(data.apps[appID], (compAddErr) => {
+		if (compAddErr) {
```

Comment:
@amaroks We probably want to do something here. Maybe fall back and cancel the creation of the app entirely? If the component can't be added to FSBL, we shouldn't add it to the launcher